transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClient.kt

```diff
@@ -96,10 +96,11 @@ interface MessagingClient {
      * Send a message to the conversation as plain text.
      *
      * @param text the plain text to send.
+     * @param customAttributes optional dictionary of attributes to send with the message. Empty by default.
      * @throws IllegalStateException
      */
     @Throws(IllegalStateException::class)
-    fun sendMessage(text: String)
+    fun sendMessage(text: String, customAttributes: Map<String, String> = emptyMap())
```

Comment:
Since default params cannot bridge to ObjC, what do you think about having two methods so that iOS isn't always required to supply `customAttributes` if they're not needed.
`fun sendMessage(text:)` and `fun sendMessage(text:customAttributes)`?

Reply:
That is interesting comment. I remember having an issue with obj-c and default arguments. However it seems not to be a case anymore. Based on how MessengerTransport.h looks like, it is capable of doing that:
``` func sendMessage(text: String, customAttributes: [String : String] = [:]) throws ``` 
Which looks legit to me.
I am looking for additional information to confirm that, and will update here if I find anything.

Reply:
Update:

"An argument whose Objective-C type is an NSDictionary defaults to nil if the NSDictionary is nullable and [:] if the NSDictionary is non-nullable, under the following conditions:

* the argument label contains the word "options" or "attributes", or the two words "user info" one after another
* the argument label is empty and the method base name ends with the word "options" or "attributes", or the two words "user info" one after another"

https://github.com/apple/swift/blob/main/docs/CToSwiftNameTranslation.md#default-argument-values


As a conclusion it is technically ok to have customAttributes as default argument. 
Having `customAttributes` as additional argument on `sendMessage()` feels more correct design-wise. Based on their requirement they should not be stored anywhere, instead they should be passed on every incoming message.